### PR TITLE
Species reconciliation for PopBio

### PR DESCRIPTION
**Pull request by bobular**
Note this merges into `eda_megastudy`

Provides a script `Load/bin/reconcilePopBioSpecies.pl` that takes the following commandline args:

```
            "gusConfigFile=s"         => \$gusConfigFile,
	    "extDbRlsSpec=s"          => \$extDbRlsSpec,
            "veupathOntologySpec=s"   => \$veupathOntologySpec,
	    "fallbackSpecies=s"       => \$fallbackSpecies,
```

`gusConfigFile` seems to be optional (presumably looking in $GUS_HOME/config/gus.config)

`extDbRlsSpec` is for the sub-study, e.g. `ISATab_VBP0000444_RSRC|b61-800a`

`veupathOntologySpec` is for the veupath ontology loaded from OBO, e.g. `OBO_Ontology_eupath_RSRC|dontcare`

`fallbackSpecies` is a prop in the dataset.xml that is usually empty, but occasionally contains a species name.

TO DO

- [x] remove old species reconcil plugin

### CODE REVIEW

File: Load/bin/reconcilePopBioSpecies.pl [hunk 1]
```diff
@@ -0,0 +1,386 @@
...
+	    "extDbRlsSpec=s"          => \$extDbRlsSpec,
+            "veupathOntologySpec=s"   => \$veupathOntologySpec,
+	    "fallbackSpecies=s"       => \$fallbackSpecies,
+            "verbose!"                => \$verbose,
+	    "testFunctions"           => \$testFunctions);  # test isAchildofB and commonAncestor functions with some hardcoded species names and quit
+
+my $gusconfig = GUS::Supported::GusConfig->new($gusConfigFile);
+
+my $db = GUS::ObjRelP::DbiDatabase->new($gusconfig->getDbiDsn(),
+                                        $gusconfig->getDatabaseLogin(),
```
Comment (jbrestel) [inline Load/bin/reconcilePopBioSpecies.pl:35]: use DBI && DBD::Oracle

set autocommit off and commit ~ 10K-20K rows

File: Load/bin/reconcilePopBioSpecies.pl [hunk 2]
```diff
@@ -0,0 +1,386 @@
...
+  ea2.atts as assay_atts
+from
+  EDA.entityattributes ea,
+  EDA.entitytype et,
+  EDA.processattributes pa,
+  EDA.entityattributes ea2,
+  EDA.entitytype et2
+where
+  ea.entity_type_id = et.entity_type_id and
+  et.name = 'sample' and
```
Comment (jbrestel) [inline Load/bin/reconcilePopBioSpecies.pl:119]: can we use the ontology term here ?

File: Load/bin/reconcilePopBioSpecies.pl [hunk 3]
```diff
@@ -0,0 +1,386 @@
...
+    die "FATAL ERROR: No species reconciliation result and no --fallbackSpecies option provided";
+  }
+
+  printf "Sample $sample_id reconciled $reconciled_species_name from [ %s ] ($qualifier)\n",
+    join ', ', sort keys %{$sample2species{$sample_id}} if $verbose;
+
+  # now add or replace the new reconciled species in the sample attributes JSON
+
+  my $sample_atts = decode_json($sample2atts_json{$sample_id});
+  $sample_atts->{$reconciled_species_variable_iri} = $reconciled_species_name;
```
Comment (jbrestel) [inline Load/bin/reconcilePopBioSpecies.pl:206]: the value here in perl should be arrayref

**Review by jbrestel — APPROVED** [review]

**Review by bobular — COMMENTED** [review]
Check for and ignore "NA" in fallbackSpecies arg

**Pull request merged**